Cargo.toml

```diff
@@ -26,5 +26,5 @@ hex = "0.4"
 
 [features]
 default = ["key_openssl_pkey"]
-key_openssl_pkey = []
-key_tpm = ["tss-esapi"]
+key_openssl_pkey = ["openssl"]
+key_tpm = ["tss-esapi", "openssl"]
```

Comment:
The `key_tpm` isn't tested in CI and fails when such a test is added with following error:
```
  --- stderr
  thread 'main' panicked at 'Failed to find tss2-sys library.: `"pkg-config" "--libs" "--cflags" "tss2-sys" "tss2-sys >= 2.3.3"` did not exit successfully: exit status: 1
  error: could not find system library 'tss2-sys' required by the 'tss-esapi-sys' crate

  --- stderr
  Package tss2-sys was not found in the pkg-config search path.
  Perhaps you should add the directory containing `tss2-sys.pc'
  to the PKG_CONFIG_PATH environment variable
  No package 'tss2-sys' found
  Package tss2-sys was not found in the pkg-config search path.
  Perhaps you should add the directory containing `tss2-sys.pc'
  to the PKG_CONFIG_PATH environment variable
  No package 'tss2-sys' found
  ', /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tss-esapi-sys-0.2.0/build.rs:41:14
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
```
this isn't related to this PR and should be fixed in a separate PR.

Reply:
I'll take a look.

Reply:
I fixed the TPM-related issues of my PR and tested it locally in a Docker container with the TPM software as described in the github workflow. Everything passes.
I'm still puzzled where this error came from. I added a test for the `key_tpm` feature. We'll see if CI throws the same error again.